iocore/net/UnixNetProcessor.cc

```diff
@@ -237,16 +237,18 @@ UnixNetProcessor::connect_re_internal(Continuation *cont, sockaddr const *target
     vc->action_ = cont;
   }
 
-  MUTEX_TRY_LOCK(lock, cont->mutex, t);
-  if (lock.is_locked()) {
-    MUTEX_TRY_LOCK(lock2, get_NetHandler(t)->mutex, t);
-    if (lock2.is_locked()) {
-      int ret;
-      ret = vc->connectUp(t, NO_FD);
-      if ((using_socks) && (ret == CONNECT_SUCCESS)) {
-        return &socksEntry->action_;
-      } else {
-        return ACTION_RESULT_DONE;
+  if (reinterpret_cast<EThread *>(this_thread()) == t) {
```

Comment:
Is this supposed to be part of this change?

Reply:
Yes, the test will not work without this change @duke8253

Reply:
I guess it seems like burying the lede here. The subject of the PR just mentions tests, but it looks like a general change in behavior here.

Reply:
So specifically are you asking me to change the commit comment?